nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -1852,6 +1853,31 @@ public Description matchEnhancedForLoop(EnhancedForLoopTree tree, VisitorState s
     if (mayBeNullExpr(state, expr)) {
       return errorBuilder.createErrorDescription(errorMessage, buildDescription(expr), state, null);
     }
+    // auto-unboxing check
+    VariableTree loopVariable = tree.getVariable();
+    Type loopVariableType = ASTHelpers.getType(loopVariable);
+    // Only relevant when the loop variable is a primitive (implies unboxing of elements).
+    if (loopVariableType == null || !loopVariableType.isPrimitive()) {
+      return Description.NO_MATCH;
+    }
+    Type expressionType = ASTHelpers.getType(expr);
+    boolean isElementNullable = false;
```

Comment:
I don't think we need this variable.  Just move the logic for reporting the error within the innermost `if`